samples/nrf9160/http_application_update/src/main.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <stdint.h>
+#include <zephyr.h>
+#include <flash.h>
+#include <bsd.h>
+#include <nrf_socket.h>
+#include <download_client.h>
+#include <logging/log.h>
+#include <gpio.h>
+#include <dfu/mcuboot.h>
+
+#define LED_PORT	LED0_GPIO_CONTROLLER
+
+static struct	device	*flash_dev;
+static bool	is_flash_page_erased[FLASH_PAGE_MAX_CNT];
+static bool	start_dfu;
+static struct	gpio_callback gpio_cb;
+static u32_t	flash_address;
+
+static int app_download_client_event_handler(struct download_client *const dfu,
+			enum download_client_evt event, u32_t status);
+
+
+static struct download_client dfu = {
+	.host = CONFIG_DOWNLOAD_HOST,
+	.resource = CONFIG_DOWNLOAD_FILE,
+	.callback = app_download_client_event_handler
+};
+
+
+/**@brief Recoverable BSD library error. */
+void bsd_recoverable_error_handler(uint32_t err)
+{
+	printk("bsdlib recoverable error: %lu\n", err);
+}
+
+/**@brief Irrecoverable BSD library error. */
+void bsd_irrecoverable_error_handler(uint32_t err)
+{
+	printk("bsdlib irrecoverable error: %lu\n", err);
```

Comment:
I'd assert in here - all in all, it's an irrecoverable error.

Reply:
I have changed this function several times already as requested by @lemrey and @gbakke. Maybe you can discuss it and come up with a common solution?

Reply:
Sigh, nevermind then...

Reply:
Added __ASSERT_NO_MSG(false); now. Same as in mqtt sample.